### ISSUE TITLE
IServerAPI.ServerIp throws NullReferenceException on DedicatedServer

### ISSUE THREAD

**Issue by Dmitry221060** [anegostudios/VintageStory-Issues#3071]
### Game Version

1.18.8, 1.18.6 (.net4)

### Platform

Windows

### Modded

None

### SP/MP

Multiplayer

### Description

Calling `IServerApi.ServerIp` on a dedicated server causes NullReferenceException. 
Calling `IServerApi.ServerIp` on a local server open for LAN doesn't change the IP from localhost. 
The underlying reason is that `Vintagestory.Server.ServerAPI.get_ServerIp()` always looks at first main socket - 
```csharp
public string ServerIp
{
	get
	{
		return this.server.mainSockets[0].LocalEndpoint;
	}
}
```
Which is a `DummyNetServer`. Dedicated server doesn't have `DummyNetServer`, and opening the game for LAN creates a new server on second socket, which should change the IP.

## Dedicated server scenario:
1. Make an dummy server-side mod that logs `IServerAPI.ServerIp` on player join.
2. Install it on a dedicated server.
3. Launch the server and try joining it.
#### Expected behavior: 
It will log the IP of the dedicated server
#### Actual behavior: 
Server will throw NullReferenceException

## Open world to LAN scenario:
1. Make an dummy server-side mod that logs `IServerAPI.ServerIp` on player join.
2. Install it on a game client.
3. Join any world.
4. Open world for LAN.
5. Join the now open server with different client.
#### Expected behavior: 
It will log:
```
...
A Client attempts connecting via Dummy connection
...
[mod log] 127.0.0.1
...
A Client attempts connecting via TCP
...
[mod log] 172.21.125.12
...
```
#### Actual behavior: 
It will log:
```
...
A Client attempts connecting via Dummy connection
...
[mod log] 127.0.0.1
...
A Client attempts connecting via TCP
...
[mod log] 127.0.0.1
...
```

### Logs

```
2.9.2023 07:34:55 [Error] Mod exception: OnPlayerNowPlayingSystem.NullReferenceException: Object reference not set to an instance of an object.
   at Vintagestory.Server.ServerAPI.get_ServerIp() in C:\Users\Tyron\Documents\vintagestory\game\VintagestoryLib\Server\API\ServerAPI.cs:line 93
   at RPVoiceChat.Server.GameServer.InitHandshake(IServerPlayer player)
   at RPVoiceChat.Server.GameServer.PlayerJoined(IServerPlayer player)
   at Vintagestory.API.Common.PlayerDelegate.Invoke(IServerPlayer byPlayer)
   at Vintagestory.Server.ServerEventManager.TriggerPlayerNowPlaying(IServerPlayer player) in C:\Users\Tyron\Documents\vintagestory\game\VintagestoryLib\Server\ServerEventManager.cs:line 649
2.9.2023 07:34:55 [Error] Exception: System.NullReferenceException: Object reference not set to an instance of an object.
   at Vintagestory.Server.ServerAPI.get_ServerIp() in C:\Users\Tyron\Documents\vintagestory\game\VintagestoryLib\Server\API\ServerAPI.cs:line 93
   at RPVoiceChat.Server.GameServer.InitHandshake(IServerPlayer player)
   at RPVoiceChat.Server.GameServer.PlayerJoined(IServerPlayer player)
   at Vintagestory.API.Common.PlayerDelegate.Invoke(IServerPlayer byPlayer)
   at Vintagestory.Server.ServerEventManager.TriggerPlayerNowPlaying(IServerPlayer player) in C:\Users\Tyron\Documents\vintagestory\game\VintagestoryLib\Server\ServerEventManager.cs:line 649
```

**Comment by Dmitry221060** [comment]
@tyronx This is a bug inside `VintagestoryLib` which isn't open source so only someone from the dev team can fix it.

**Comment by tyronx** [comment]
thanks for the report, the crash will be fixed in 1.19.

It will not show your remote ip though, just the ip the server is listening at.
